### PR TITLE
[RUM-11854]: Account id  and User id propagated in baggage header

### PR DESCRIPTION
**Pull request by satween**
### What does this PR do?

Adds support for `accountId` and `userId` propagation through `baggage` header.
Add and refactor tests that is related to rum context propagation

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](../CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-trace-internal/src/test/kotlin/com/datadog/trace/core/propagation/DatadogHttpCodecTest.kt [hunk 1]
```diff
@@ -71,7 +79,7 @@ internal class DatadogHttpCodecTest {
                 capture()
             )
 
-            assertThat(firstValue).isEqualTo("session.id=$fakeSessionId")
+            assertThat(firstValue).isEqualTo("account.id=$fakeAccountId,user.id=$fakeUserId,session.id=$fakeSessionId")
```
Comment (ambushwork) [inline features/dd-sdk-android-trace-internal/src/test/kotlin/com/datadog/trace/core/propagation/DatadogHttpCodecTest.kt]: I think it's also nice to test  the case when `accountId` or `userId` is null or empty, we don't have the string like "account.id=,user.id=,"

Comment (satween) [reply features/dd-sdk-android-trace-internal/src/test/kotlin/com/datadog/trace/core/propagation/DatadogHttpCodecTest.kt]: yep, make sense, added
